app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx

```diff
@@ -54,10 +60,19 @@ const displaySelectOptionLabels = (options, values) => {
 
 const ExperimentDetails: React.FC = () => {
   const experimentData = useExperimentState();
+  const visualizationRef = React.useRef(null);
   const dispatch = useExperimentDispatch();
 
   const handleClickRequestApproval = async () => dispatch(requestReview());
 
+  React.useEffect(() => {
+    if (!visualizationRef.current || !window.initSvelte) {
+      return;
+    }
+
+    window.initSvelte(visualizationRef.current);
+  });
```

Comment:
Oh actually I think if you pass as the second arg `[visualizationRef]` then that will achieve the desired effect since it'll fire the effect whenever that changes, so then i think it looks something like

```js
  React.useEffect(() => {
    if (visualizationRef) {
    window.initSvelte(visualizationRef.current);
    }
  }, [visualizationRef]);
```

Reply:
Actually, I tried it with both `[visualizationRef]` and `[visualizationRef.current]` as the second arg and neither seem to work. They both result in only one call to `useEffect()` in which `visualizationRef.current` is still `undefined`

Reply:
Update: `[visualizationRef.current]` works but it takes like 30 seconds or more until `useEffect()` is called a second time.

Reply:
Oh yikes what the heck

Reply:
Oh it's because 30s is the timeout to refetch from the API which is triggering a re render.  Yeah I tried locally and got exactly what you're describing, okay the way you have it written totally works I vote just keep it like that unless @LZoog knows something better?

Reply:
Is this because we're using React _and_ Svelte? Or would we still see this 30s penalty box if we were just using React?

Reply:
@pdehaan it's what I just said above ^

Reply:
Apologies for the late response here, I missed the @ mention😩

```
  React.useEffect(() => {
    if (visualizationRef.current) {
      window.initSvelte(visualizationRef.current);
    }
  }, [visualizationRef]);
```
This is what I would have suggested, if this doesn't work as expected then I think this is fine, we'll review this when we get to it in `nimbus-ui` anyway 🙂

Reply:
Yeah so I did some more digging @LZoog  

It's not `visualizationRef` since that's an object that gets returned from the `useRef` call, what changes is the `visualizationRef.current` value inside, and if you set that as the useEffect trigger, it "works" in the sense that 

- On the first pass rendering the component, `visualizationRef.current` is `null`, the if statement fails, and it passes over, then it finishes its first render, which adds the `div` and updates the ref
- Now it's done rendering, and nothing triggers it to render again so it just sets there
- After 30s, the API fetch timeout happens, which updates state, which triggers a re render, and now the next pass through when it hits `useEffect` the value has changed, so it calls again, passes the if check, and inits svelte

So I think useEffect doesn't actively 'listen' for changes to the value, it only compares them when something else triggers a render, so for the way you wrote it to work, we'd have to trigger a second render after the first render places the div, which I wasn't sure how to do.

Anyways yes it works the way @emtwo wrote it so, no reason not to keep it that way 👍